tests/core/test_session.py

```diff
@@ -947,3 +951,41 @@ async def test_set_transaction_mix_multiple_devices():
 
     assert (await session_a.get("a")).value == 1
     assert (await session_b.get("a")).value == 2
+
+
+@pytest.mark.asyncio()
+async def test_ensure_compatibility_ok():
```

Comment:
I think this one should be a test in the mock directory.

However, I agree that such a test is necessary here. I would prefer to have the version tested against specifically written in the test (not depending on the version the mock uses), similar to the other tests. 
With version Max_tested_version, there is no error

Reply:
Good idea. I moved this test to the hwmock test and adapted the following test also to test ok versions.